pkg/scheduler/objects/application.go

```diff
@@ -1520,3 +1533,19 @@ func (sa *Application) GetRejectedMessage() string {
 	defer sa.RUnlock()
 	return sa.rejectedMessage
 }
+
+func (sa *Application) SetPlaceholderData(taskGroupName string, minResource *resources.Resource, requiredNode string) {
```

Comment:
This should be called as part of adding the ask in the application:
* unlocked function
* internal function (not exported)
We need an exported and locked `GetPlaceholdersData() []*PlaceholderData` to call from the web service like the `GetAllAllocations()` implementation